contracts/6/insta-swaps/LiquidityPoolManager.sol

```diff
@@ -0,0 +1,163 @@
+        require(liquidityProviderMap[NATIVE][_msgSender()] >= amount, "Not enough balance");
+        liquidityProviderMap[NATIVE][_msgSender()] = liquidityProviderMap[NATIVE][_msgSender()].sub(amount);
+        tokenBalanceMap[NATIVE] = tokenBalanceMap[NATIVE].sub(amount);
+        require(_msgSender().send(amount), "Native Transfer Failed");
+    }
+
+    function addTokenLiquidity( address tokenAddress, uint256 amount ) public tokenChecks(tokenAddress) whenNotPaused {
+        require(amount > 0, "amount should be greater then 0");
+
+        IERC20 erc20 = IERC20(tokenAddress);
+
+        require(erc20.transferFrom(_msgSender(),address(this),amount),"Token Transfer Failed");
+
+        liquidityProviderMap[tokenAddress][_msgSender()] = liquidityProviderMap[tokenAddress][_msgSender()].add(amount);
+        tokenBalanceMap[tokenAddress] = tokenBalanceMap[tokenAddress].add(amount);
+    }
+
+    function removeTokenLiquidity( address tokenAddress, uint256 amount ) public tokenChecks(tokenAddress) whenNotPaused {
+        require(amount > 0, "amount should be greater then 0");
+        require(liquidityProviderMap[tokenAddress][_msgSender()] >= amount, "Not enough balance");
+
+        IERC20 erc20 = IERC20(tokenAddress);
+        require(erc20.transfer(_msgSender(),amount),"Token Transfer Failed");
+
+        liquidityProviderMap[tokenAddress][_msgSender()] = liquidityProviderMap[tokenAddress][_msgSender()].sub(amount);
+        tokenBalanceMap[tokenAddress] = tokenBalanceMap[tokenAddress].sub(amount);
+    }
+
+    function deposit( address tokenAddress, address receiver, uint256 amount ) public tokenChecks(tokenAddress) onlyExecutorOrOwner whenNotPaused {
+        require(tokenCapMap[tokenAddress] == 0 || tokenCapMap[tokenAddress] >= amount,"Deposit amount exceeds allowed Cap limit");
+        require(receiver != address(0), "Receiver address cannot be 0");
+        require(amount > 0, "amount should be greater then 0");
+
+        IERC20 erc20 = IERC20(tokenAddress);
+
+        require(erc20.transferFrom(_msgSender(),address(this),amount),"Deposit Failed");
+
+        emit Deposit(tokenAddress, receiver, amount);
+    }
+
+    function sendFundsToUser( address tokenAddress, uint256 amount, address payable receiver, string memory depositHash, uint256 gasFees ) public nonReentrant onlyExecutorOrOwner tokenChecks(tokenAddress) whenNotPaused {
+        require(tokenCapMap[tokenAddress] == 0 || tokenCapMap[tokenAddress] >= amount, "Withdraw amount exceeds allowed Cap limit");
+        require(receiver != address(0), "Bad receiver address");
+
+        bytes32 hashSendTransaction = keccak256(
+            abi.encode(
+                tokenAddress,
+                amount,
+                receiver,
+                keccak256(bytes(depositHash))
+            )
+        );
+
+        require(!sendFundsMap[hashSendTransaction],"Already Processed");
+
+        uint256 bcnmyCommission = amount.mul(biconomyCommission).div(100);
```

Comment:
Let use 10,000 as basis points. 
So whatever percentage value we want to add, we multiple that by 100 and use that value.
eg for 50% commision we use 50*100 value